[CostModel] Handle all cost kinds in getCmpSelInstrCost

davemgreen opened the pull request:
Currently we always produce a cost of 1 for all CostKinds that are not RecipThroughput, which can underestimate the cost if the type has a higher legalization cost (like larger vectors). This relaxes it to cover all cost kinds.

RKSimon commented on llvm/include/llvm/CodeGen/BasicTTIImpl.h:
```diff
@@ -1369,8 +1369,7 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
 
-    // TODO: Handle other cost kinds.
-    if (CostKind != TTI::TCK_RecipThroughput)
+    if (getTLI()->getValueType(DL, ValTy, true) == MVT::Other)
```
where does the MVT::Other case occur?

RKSimon commented:
merge to trunk to see if those CI failures were intermittent?

davemgreen committed:
[CostModel] Handle all cost kinds in getCmpSelInstrCost

Currently we always produce a cost of 1 for all CostKinds that are not
RecipThroughput, which can underestimate the cost if the type has a higher
legalization cost (like larger vectors). This relaxes it to cover all cost
kinds.

davemgreen commented:
> merge to trunk to see if those CI failures were intermittent?

Updated. It seemed to run OK locally.

RKSimon reviewed (approved):
LGTM - as a side note we should really try to cleanup more CostModel tests to use "-cost-kind=all"

c-rhodes reviewed (approved):
also LGTM cheers
